### PR TITLE
Add config s3 client with different bucket lookup type

### PR DESCRIPTION
**Pull request by guangbochen**
**Problem:**
RKE S3 Backup doesn't work with Aliyun S3

**Solution:**
Aliyun OSS is s3 API compatible but it requires a different type of URL lookup, added auto detect of `BucketLookup` type by checking the endpoint address.

**Issues:**
https://github.com/rancher/rancher/issues/18262

**Related PRs:**
https://github.com/rancher/rancher/pull/18329

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

**Comment by alena1108** [comment]
@guangbochen lets make sure the pr doesn't break AWS S3 and Minio

**Comment by guangbochen** [comment]
Verified on `Minio`, `AWS S3`, `Google GCS`, `Aliyun OSS`, `AWS S3 with IAM Role`.
image
image

File: main.go [hunk 1]
```diff
@@ -447,18 +447,25 @@ func setS3Service(bc *backupConfig, useSSL bool) (*minio.Client, error) {
...
 		// if the s3 access key and secret is not set use iam role
 		if len(bc.AccessKey) == 0 && len(bc.SecretKey) == 0 {
 			log.Info("invoking set s3 service client use IAM role")
-			iam := credentials.NewIAM("")
-			svc, err = minio.NewWithCredentials("s3.amazonaws.com", iam, true, "")
-		} else if len(bc.Region) != 0 {
-			svc, err = minio.NewWithRegion(bc.Endpoint, bc.AccessKey, bc.SecretKey, useSSL, bc.Region)
+			cred = credentials.NewIAM("")
+			if bc.Endpoint == "" {
+				bc.Endpoint = "s3.amazonaws.com"
```
Comment (moelsayed) [inline main.go]: Can we but this in a constant ?

**Comment by moelsayed** [comment]
@guangbochen can you please check the code here https://github.com/rancher/rancher/blob/master/pkg/controllers/management/etcdbackup/etcdbackup.go#L234 and make required updates as well?

**Comment by guangbochen** [comment]
@moelsayed sure will check now, thanks.

**Comment by guangbochen** [comment]
@moelsayed this one is updated and can u please also help to review the rancher/rancher#18329, thanks. cc @alena1108.

**Pull request merged**